lib/rules/brace-style.js

```diff
@@ -2,37 +2,171 @@
+  }
+
+  // Functions, Mixins, Loops
+  if (node.is('atrule') || node.is('mixin') || node.is('loop')) {
+    return node.contains('atkeyword') ? node.last('atkeyword') : false;
+  }
+};
+
+/**
+ * Determine if current node is an exception and end checks if it is
+ *
+ * @param {Object} node - The original node
+ * @param {object} currentNode - The current node block
+ * @param {Object} previousNode - The node previous to our current node
+ * @returns {bool} Wether or not the it is an exception
+ */
+var isException = function (node, currentNode, previousNode) {
+  // If we've picked up a return @rule ignore it
+  if (node.is('atrule')) {
+    if (previousNode.contains('ident') && previousNode.first('ident').content === 'return') {
+      return true;
+    }
+  }
+  return false;
+};
+
+/**
+ * Determine if statement is a single line statement
+ *
+ * @param {Object} node - The statement to check
+ * @returns {bool} True or false
+ */
 var isSingleLineStatement = function (node) {
   return node.start.line === node.end.line;
 };
 
-var createPreviousNode = function (node) {
-  return {
-    content: '',
-    start: {
-      line: node.start.line,
-      column: node.start.column - 1
-    },
-    end: {
-      line: node.end.line,
-      column: node.end.column
+/**
+ * Determine if opening brace of statement is on a new line
+ *
+ * @param {Object} nodeA - The previous block
+ * @param {Object} nodeB - The current block
+ * @returns {bool} True or false
+ */
+var isOpeningBraceOnNewLine = function (nodeA, nodeB) {
+  return nodeA.end.line === nodeB.start.line;
+};
+
+/**
+ * Determine if closing brace of statement is on new line
+ *
+ * @param {Object} node - The current block
+ * @returns {bool|null} True or false if relevant else null
+ */
+var isClosingBraceOnNewLine = function (node) {
+  if (node.contains('block')) {
+    var content = node.first('block'),
+        contentLength = content.length - 1,
+        lastNode = content.get(contentLength);
+
+    if (lastNode.is('space') && helpers.hasEOL(lastNode.content)) {
+      return true;
     }
-  };
+    return false;
+  }
+  return null;
 };
 
-var getElsePrevious = function (atKeyword, parent, i) {
-  if (atKeyword.contains('ident')) {
-    if (atKeyword.first('ident').content === 'else') {
-      var prev = parent.get(i - 1);
+/**
+ * Determine if condition starts on a new line by checking the leading node for
+ * an end-of-line
+ *
+ * @param {Object} node - The node that is our condition
+ * @param {Object} parentNode - The condition node's parent
+ * @param {Number} j - The index of our node in the context of the parent's children
+ * @returns {bool|null} True or false if relevant else null
+ */
+var isConditionOnNewLine = function (node, parentNode, j) {
+  // Only check if it's an @else condition
+  if (node.contains('ident') && node.first('ident').content === 'else') {
+    // Reverse back up tree
+    var previousChild = parentNode.get(--j);
```

Comment:
This might not be a node so we should either do `|| false` or we should check it exists on the next line